components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -793,25 +824,37 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
                     String tokenType = resultSet.getString(10);
                     String refreshToken = resultSet.getString(11);
                     String tokenId = resultSet.getString(12);
+                    String grantType = resultSet.getString(13);
+
                     User user = new User();
                     user.setUserName(authorizedUser);
                     user.setUserStoreDomain(userDomain);
                     user.setTenantDomain(tenantDomain);
+
                     dataDO = new AccessTokenDO(consumerKey, user, scope, issuedTime, refreshTokenIssuedTime,
                             validityPeriodInMillis, refreshTokenValidityPeriodMillis, tokenType);
                     dataDO.setAccessToken(accessTokenIdentifier);
                     dataDO.setRefreshToken(refreshToken);
                     dataDO.setTokenId(tokenId);
-                }else{
+                    dataDO.setGrantType(grantType);
+
+                } else {
                     scopes.add(resultSet.getString(5));
                 }
+
                 iterateId++;
             }
+
             if (scopes.size() > 0 && dataDO != null){
                 dataDO.setScope((String[])ArrayUtils.addAll(dataDO.getScope(), 
                      scopes.toArray(new String[scopes.size()])));
             }
+
             connection.commit();
+
+        } catch (IdentityException e) {
```

Comment:
IdentityException should not be throw here. Please check. IdentityDatabaseUtil.getDBConnection() now throws a IdentityRuntimeException which you don't need to handle